components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/validators/AlphanumericValidator.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.input.validation.mgt.model.validators;
+
+import org.wso2.carbon.identity.input.validation.mgt.exceptions.InputValidationMgtClientException;
+import org.wso2.carbon.identity.input.validation.mgt.model.Property;
+import org.wso2.carbon.identity.input.validation.mgt.model.ValidationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.DEFAULT_ALPHANUMERIC_REGEX_PATTERN;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.ENABLE;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.USERNAME;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_VALIDATION_ALPHANUMERIC_FORMAT_MISMATCH;
+
+/**
+ * Alphanumeric validator.
+ */
+public class AlphanumericValidator extends AbstractRulesValidator {
+
+    private final List<String> allowedFields = new ArrayList<String>() {{
+        add(USERNAME);
+    }};
+
+    @Override
+    public boolean isAllowedField(String field) {
+
+        return allowedFields.contains(field);
+    }
+
+    /**
+     * Validate the string against the validation criteria.
+     *
+     * @param context   Validation Context.
+     * @return boolean
+     * @throws InputValidationMgtClientException Error when string does not satisfy the validation criteria
+     */
+    @Override
+    public boolean validate(ValidationContext context) throws InputValidationMgtClientException {
+
+        String value = context.getValue();
+        String field = context.getField();
+        Map<String, String> attributesMap = context.getProperties();
+        String alphanumericRegEx = DEFAULT_ALPHANUMERIC_REGEX_PATTERN;
+
+        // Check whether value satisfies the alphanumeric criteria.
+        if (attributesMap.containsKey(ENABLE)) {
+            if (Boolean.parseBoolean(attributesMap.get(ENABLE)) && value != null && !value.matches(alphanumericRegEx)) {
```

Comment:
Is it possible to become the 'value' null or attributes map does not contain the 'enabled' property? If so we need to throw correct errors. In the logic that part is not handled.

Reply:
Addressed with commit 6e4ccba6376101562c801f735b660c94f4a280b2